### PR TITLE
documentation - add ContextShift needed for MVar.empty and IO.start

### PR DESCRIPTION
**Pull request by prayagupd**
add import for `ContextShift`, 

```bash
[error] Building this implicit value might depend on having an implicit
[error] s.c.ExecutionContext in scope, a Scheduler, a ContextShift[cats.effect.IO]
[error] or some equivalent type.
[error]       channel: MVar[IO, Option[Int]] <- MVar[IO].empty[Option[Int]]
```

### CODE REVIEW

File: site/src/main/tut/concurrency/mvar.md [hunk 1]
```diff
@@ -160,6 +160,13 @@ def consumer(ch: Channel[Int], sum: Long): IO[Long] =
       IO.pure(sum) // we are done!
   }
 
+// ContextShift required for
+// 1) MVar.empty
+// 2) IO.start
+// for ContextShift documentation see https://typelevel.org/cats-effect/datatypes/contextshift.html
```
[the remark this answers is not in the document]
Comment (Daenyth) [reply site/src/main/tut/concurrency/mvar.md]: As someone training a team of mixed-skill-level and mixed-scala-experience developers, I can say that cross-links like this are really useful.

[the remark this answers is not in the document]
Comment (prayagupd) [reply site/src/main/tut/concurrency/mvar.md]: Hi @alexandru I am not a big fan of comments either as they might to be changed along with
codebase. Better when code speaks for itself but in this case comment is to help understand
where `ContextShift` is used. 

I thought It helps when I directly jump into `MVar` concept even with little knowledge of cats-effect and scala. 

Let me know I can remove the comment if you think other ways :)

**Comment by prayagupd** [comment]
Hi @rossabaker There are few other code failing in tutorials but `tut` looks fine on `mvar.md`.

```bash
sbt
project microsite

sbt:microsite> tut
  [info] Running tut.TutMain /Users/prayagupd/cats-effect/site/src/main/tut /Users/prayagupd/cats-effect/site/target/scala-2.12/resource_managed/main/jekyll .*\.(md|markdown|txt|htm|html) -language:_ -deprecation -encoding UTF-8 -feature -unchecked -Ypartial-unification -Xplugin:/Users/a1353612/.coursier/cache/v1/https/repo1.maven.org/maven2/org/spire-math/kind-projector_2.12/0.9.9/kind-projector_2.12-0.9.9.jar
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/tutorial/tutorial.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/contextshift.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/fiber.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/syncio.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/index.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/resource.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/ioapp.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/timer.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/io.md
  [tut] *** Error reported at /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/io.md:1220
<console>:70: error: could not find implicit value for parameter p: cats.NonEmptyParallel[cats.effect.IO,F]
  val program = (ioA, ioB, ioC).parMapN { (_, _, _) => () }
  ^
  [tut] *** Error reported at /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/io.md:1252
  <console>:71: error: could not find implicit value for parameter p: cats.NonEmptyParallel[cats.effect.IO,F]
    (ioA, ioB).parMapN((_, _) => ())
    ^
    [tut] *** Error reported at /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/io.md:1267
    <console>:79: error: could not find implicit value for parameter P: cats.Parallel[cats.effect.IO,F]
      val ioOfList = aLotOfIOs.parSequence
      ^
      [tut] *** Error reported at /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/io.md:1279
      <console>:78: error: could not find implicit value for parameter P: cats.Parallel[cats.effect.IO,F]
        val results = NonEmptyList.of(1, 2, 3).parTraverse { i =>
        ^
          [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/clock.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/concurrency/ref.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/concurrency/basics.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/concurrency/deferred.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/concurrency/mvar.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/concurrency/index.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/concurrency/semaphore.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/effect.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/bracket.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/liftio.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/async.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/index.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/concurrent-effect.md
  [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/sync.md
  [tut] *** Error reported at /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/sync.md:17
<console>:14: error: not found: type Bracket
  trait Sync[F[_]] extends Bracket[F, Throwable] with Defer[F] {
  ^
  <console>:14: error: not found: type Defer
    trait Sync[F[_]] extends Bracket[F, Throwable] with Defer[F] {
    ^
    <console>:16: error: not found: value pure
      def delay[A](thunk: => A): F[A] = suspend(pure(thunk))
      ^
      [tut] compiling: /Users/prayagupd/cats-effect/site/src/main/tut/typeclasses/concurrent.md
      [error] (run-main-3) tut.TutException: Tut execution failed.
      [error] tut.TutException: Tut execution failed.
      [error] 	at tut.TutMain$.$anonfun$runl$6(TutMain.scala:17)
      [error] 	at tut.felix.IO$Compute.loop$1(IO.scala:121)
      [error] 	at tut.felix.IO$Compute.unsafePerformIO(IO.scala:125)
      [error] 	at tut.TutMain$.main(TutMain.scala:8)
      [error] 	at tut.TutMain.main(TutMain.scala)
      [error] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
      [error] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
      [error] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
      [error] 	at java.lang.reflect.Method.invoke(Method.java:498)
      [error] 	at sbt.Run.invokeMain(Run.scala:98)
      [error] 	at sbt.Run.run0(Run.scala:92)
      [error] 	at sbt.Run.execute$1(Run.scala:68)
      [error] 	at sbt.Run.$anonfun$run$4(Run.scala:80)
      [error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
      [error] 	at sbt.util.InterfaceUtil$$anon$1.get(InterfaceUtil.scala:10)
      [error] 	at sbt.TrapExit$App.run(TrapExit.scala:253)
      [error] 	at java.lang.Thread.run(Thread.java:748)
      [error] java.lang.RuntimeException: Nonzero exit code: 1
      [error] 	at scala.sys.package$.error(package.scala:26)
      [error] 	at tut.TutPlugin$.$anonfun$tutOne$1(TutPlugin.scala:149)
      [error] 	at scala.util.Success.foreach(Try.scala:249)
      [error] 	at tut.TutPlugin$.tutOne(TutPlugin.scala:149)
      [error] 	at tut.TutPlugin$.$anonfun$projectSettings$19(TutPlugin.scala:72)
      [error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
      [error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:40)
      [error] 	at sbt.std.Transform$$anon$4.work(System.scala:67)
      [error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:269)
      [error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
      [error] 	at sbt.Execute.work(Execute.scala:278)
      [error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:269)
      [error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
      [error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
      [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
      [error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
      [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
      [error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
      [error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
      [error] 	at java.lang.Thread.run(Thread.java:748)
      [error] (tut) Nonzero exit code: 1
      [error] Total time: 50 s, completed Jan 12, 2019 1:01:02 PM
```

**Comment by rossabaker** [comment]
I think we're not seeing these errors in Travis because of #471.

**Comment by prayagupd** [comment]
I see. 
Different topic but one thing I did not understand is in `io.md` (https://github.com/typelevel/cats-effect/blob/master/site/src/main/tut/datatypes/io.md#parmapn) though there is `ContexShift` available `tut` complains about it. I could just copy paste the code and able to run as well.

Error 

```bash
$ sbt microsite/tut

[tut] *** Error reported at /Users/prayagupd/cats-effect/site/src/main/tut/datatypes/io.md:1220
<console>:70: error: could not find implicit value for parameter p: cats.NonEmptyParallel[cats.effect.IO,F]
       val program = (ioA, ioB, ioC).parMapN { (_, _, _) => () }
                                             ^
```

while code looks good to me, 

```scala
1210 ```tut:silent                                                                                       
1211 import cats.syntax.all._                                                                            
1212 import scala.concurrent.ExecutionContext                                                            
1213 import cats.effect.ContextShift                                                                     
1214                                                                                                     
1215 val ioA = IO(println("Running ioA"))                                                                
1216 val ioB = IO(println("Running ioB"))                                                                
1217 val ioC = IO(println("Running ioC"))                                                                
1218                                                                                                     
1219 implicit val ctxShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)                                                                
1220 val program = (ioA, ioB, ioC).parMapN { (_, _, _) => () }                                           
1221                                                                                                     
1222 program.unsafeRunSync()                                                                             
1223 //=> Running ioB                                                                                    
1224 //=> Running ioC                                                                                    
1225 //=> Running ioA                                                                                    
1226 ```
```

**Comment by alexandru** [comment]
Please merge with master and see why the build fails.

**Comment by djspiewak** [comment]
Sorry this took so long for me to loop back to!

**Pull request merged**